File: DSA/Ayush_Mishra_590014844/Day_5_Question_1.c
#include <stdio.h>

int findSingle(int nums[], int size) {
    int result = 0;
    for (int i = 0; i < size; i++) {
        result ^= nums[i];  // XOR logic
    }
    return result;
}

int main() {
    int nums[] = {4, 1, 2, 1, 2};
    int size = sizeof(nums) / sizeof(nums[0]);
    printf("Unique element: %d\n", findSingle(nums, size));
    return 0;
}

File: DSA/VenikaPanwar_590013755/Day_5_Question_1.java
import java.util.*;

public class Day_5_Question_1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter array in format: {2 2 1}");
        String input = sc.nextLine();
        input = input.replace("{", "").replace("}", "").trim();
        String[] parts = input.split(" ");
        int[] nums = new int[parts.length];

        for (int i = 0; i < parts.length; i++) {
            nums[i] = Integer.parseInt(parts[i]);
        }

        int unique = 0;
        for (int i = 0; i < nums.length; i++) {
            unique = unique ^ nums[i]; // XOR: cancels out duplicates
        }

        System.out.println("Unique Element is: " + unique);
    }
}
